lib/rules/spaced-comment.js

```diff
@@ -105,30 +114,45 @@ function createAlwaysStylePattern(markers, exceptions) {
             pattern += escapeAndRepeat(exceptions[0]);
         } else {
 
-            // a sequence of one of exception patterns.
+            // a sequence of one of the exception patterns.
             pattern += "(?:";
             pattern += exceptions.map(escapeAndRepeat).join("|");
             pattern += ")";
         }
-
-        pattern += "(?:$|[\n\r]))"; // the sequence continues until the end.
+        if (location === "begin") {
+            pattern += "(?:$|[\n\r]))"; // sequence continues until the end
+        } else {
+            pattern += ")$";
+        }
     }
 
     return new RegExp(pattern);
 }
 
 /**
  * Creates RegExp object for `never` mode.
- * Generated pattern is below:
+ * Generated pattern for beginning of comment:
  *
  * 1. First, a marker or nothing (captured).
  * 2. Next, a space or a tab.
  *
+ * Generated pattern for end of comment:
+ *
+ * 1. A space or tab.
+ * 2. End of comment.
+ *
  * @param {string[]} markers - A marker list.
+ * @param {string} location - Specifies if pattern checks the beginning or end of a comment.
  * @returns {RegExp} A RegExp object for `never` mode.
  */
-function createNeverStylePattern(markers) {
-    var pattern = "^(" + markers.map(escape).join("|") + ")?[ \t]+";
+function createNeverStylePattern(markers, location) {
```

Comment:
Seems like you shouldn't add a location parameter to this function. It's simple enough to create the markerless version of the never pattern without resorting to passing an empty array here.

My advice would be to leave this function as is and instead inline `/[ \t]+$/` in the ternary for endRegex